### PR TITLE
Reduce yaml schedule files for guided_xfs cases

### PR DESCRIPTION
**Pull request by chcao**
Reduce yaml schedule files for guided_xfs cases.

- Related ticket: https://progress.opensuse.org/issues/155911
- Needles: n/a
- Verification run: 
- https://openqa.suse.de/tests/14931643 (x86_64)
- https://openqa.suse.de/tests/14932106 (x86_64 svirt-xen-hvm)
- https://openqa.suse.de/tests/14910576 (x86_64 svirt-xen-pv)
- https://openqa.suse.de/tests/14932103 (s390x kvm)
- https://openqa.suse.de/tests/14931644 (s390x zvm)
- https://openqa.suse.de/tests/14910577 (ppc64le)
- https://openqa.suse.de/tests/14932104 (aarch64)
- Related mr: https://gitlab.suse.de/qe-yam/openqa-job-groups/-/merge_requests/246

### CODE REVIEW

File: tests/console/validate_partition_table.pm [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+use scheduler 'get_test_suite_data';
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
+    my $disks = get_test_suite_data()->{disks};
+
+    my $actual;
+    my $table_type_parted = get_test_suite_data()->{table_type_parted};
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:24]: test data is for validate against that data and compare it with the current data from the system, in general not for providing conditions (in a similar way that we would do with settings, we try to avoid that).
The problem here is that we don't have always available blkid in all the systems, in particular in s390x.
I hope that my advice here doesn't contradict what I told you previous PR (but I can see each case different unfortunately). Things that comes to my mind...

- can be installed blkid in s390x? if so, it would be straightforward, we are not force to use any particular command and the command to use is not part of functional testing, any option is ok.
- in one line we could check if the tool is installed in the system, rpm -q or similar (there are example in the repo) and if not present, fall back to the other tool.

I'm not sure why is that and if it is only an s390x thingy, it might be, but perhaps we could find this in other systems and we don't knot the reason, that is why I would not like to have and `if s390x` because we don't know it only applies to it.

In summary you can just keep the very same code that run in the two existing modules that you borrowed below, but do some query before it to know which function you need to call, please store the function in some variable that then you call it later, instead doing the logic inside the loop.

Comment (chcao) [reply tests/console/validate_partition_table.pm:24]: blkid is already installed in s390x, the problem is s390x zvm is using `dasda`, and there's no output when doing 
`blkid /dev/dasda`, I already asked developer whether it is by design or not.

Comment (jknphy) [reply tests/console/validate_partition_table.pm:24]: afir is by design.

Comment (jknphy) [reply tests/console/validate_partition_table.pm:24]: ok, then knowing the reason is fine to have the condition in the code, we already have test data for it,
if `table_type eq 'dasd'` pick a tool if not the other. You can add a comment about not having uuid and not being able to use the other tool, that should be very useful.

Comment (chcao) [reply tests/console/validate_partition_table.pm]: updated, thanks.

File: tests/console/validate_partition_table.pm [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# SUSE's openQA tests
+#
+# Copyright 2021 SUSE LLC
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:3]: 2024

File: tests/console/validate_partition_table.pm [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
+    my $disks = get_test_suite_data()->{disks};
+
+    my $actual;
+    # On s390x zvm, the disk is using dasd, /dev/dasda hasn't uuid,
+    # so use parted to check its disks; for other arches, use blkid.
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:25]: this comment should be above line 28

File: tests/console/validate_partition_table.pm [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+use scheduler 'get_test_suite_data';
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
+    my $disks = get_test_suite_data()->{disks};
+
+    my $actual;
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:23]: you only use $actual internally, you can declare it inside the foreach.

**Review by jknphy — APPROVED** [review]

File: tests/console/validate_partition_table.pm [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+use base "opensusebasetest";
+use testapi;
+
+use scheduler 'get_test_suite_data';
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
+    my $disks = get_test_suite_data()->{disks};
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:21]: it is better to avoid too many assignation (for maintenability), you can use this inside the foreach.

File: tests/console/validate_partition_table.pm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+use warnings;
+use base "opensusebasetest";
+use testapi;
+
+use scheduler 'get_test_suite_data';
+use filesystem_utils qw(get_partition_table_via_blkid partition_table);
+
+sub run {
+    select_console('root-console');
+    my $errors = '';
```
Comment (jknphy) [inline tests/console/validate_partition_table.pm:20]: just for code aesthetics, I would expect this variable to be just about the foreach and one empty line to separate it from the select_console.

Comment (chcao) [reply tests/console/validate_partition_table.pm]: all updated, thanks.
